src/AppInstallerCLICore/VTSupport.h

```diff
@@ -133,6 +134,20 @@ namespace AppInstaller::CLI::VirtualTerminal
         extern const Sequence EraseLineBackward;
         extern const Sequence EraseLineEntirely;
     }
+
+    namespace Progress
+    {
+        enum class ProgressState
+        {
+            None,
+            Indeterminate,
+            Normal,
+            Paused,
+            Error
+        };
+
+        ConstructedSequence ProgressBar(ProgressState state, std::optional<int> percentage = std::nullopt);

```

Comment:
```suggestion
        ConstructedSequence Construct(State state, std::optional<uint32_t> percentage = std::nullopt);
```

To prevent confusion with `ProgressBar` in the `ExecutionProgress.cpp`, and reduce repetition again.  Also, since the value cannot be negative, let's just use a type without the range.

Reply:
Applied your suggestion in c4d5c48a1c5a638d15025687a060f752bd5be3f0.